### ISSUE TITLE
Random user agent shield

### ISSUE THREAD

**Issue by gerardo-junior** [brave/brave-browser#5420]
To decrease tracking power by fingerprint a good idea is to add a new shield with the same logic as [Random User-Agent](https://chrome.google.com/webstore/detail/random-user-agent/einpaelgookohagofgnnkcfjbkkgepnp) (*[github](https://github.com/tarampampam/random-user-agent)) that a gen a new User-Agent by request or time.

**Comment by gerardo-junior** [comment]
@tomlowenthal just for personal doubt, why is this not actionable? Is it too hard to implement? Or does it have no real impact on privacy?

thank you, :)

**Comment by tomlowenthal** [comment]
We already have a clear plan for Brave's user agent string. Our goal is to have a consistent anonymity set of Brave browsers which are indistinguishable, without triggering any compatibility problems with sites that parse UA strings poorly. Randomizing the string has the potential to cause those compatibility problems and doesn't actually hide you better — it makes you particularly distinctive for the duration of each random string's usage.

**Comment by bsclifton** [comment]
@gerardo-junior I believe there are extensions (which do work) for changing user-agent. So this could be a great idea for an extension (just not a feature we include)

**Comment by gerardo-junior** [comment]
@bsclifton as I put in the issue there is the "random-user-agent" but for security and performace reasons I suggested this build-in because if setting it to ["last 2 version", "not dead", " windows> = 8 "] or something I have no big problems with the render of common pages. easylist's with brave native ad/track blocker gives a lot more trouble from my experience. and I found it interesting to look more like the crowd especially as brave is not so popular yet. anyway, i understand and agree the reasons of @tomlowenthal
